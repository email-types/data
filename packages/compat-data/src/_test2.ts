export const data = {
  products: [
    {
      name: 'Outlook',
      clients: [
        {
          name: 'Windows Live Mail 2009-2012',
          succeeds: 'Windows Mail',
          kind: 'application',
          os: 'windows',
          engine: 'trident',
          versions: [9.0, 10.0, 11.0, 12.0, 14.0, 15.0, 16.0],
        },

        {
          name: 'Windows Live Mail 2009-2012',
          succeeds: 'Windows Mail',
          kind: 'application',
          os: 'windows',
          engine: 'trident',
          versions: [2010, 2011, 2012],
        },
      ],
    },
  ],
};
