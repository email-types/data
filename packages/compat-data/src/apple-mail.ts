export const data = {
  clients: {
    name: 'Apple Mail',
    aliases: [],
    prefix: 'Webkit',

    versions: [
      {
        version: 13.0,
        usage: 0.0,
        prefix: 'Webkit',
      },
      {
        version: 12.0,
        usage: 0.0,
        prefix: 'Webkit',
      },
      {
        version: 11.0,
        usage: 0.0,
        prefix: 'Webkit',
      },
      {
        version: 9.3,
        usage: 0.0,
        prefix: 'Webkit',
      },
      {
        version: 7.3,
        usage: 0.0,
        prefix: 'Webkit',
      },
    ],
  },
};
