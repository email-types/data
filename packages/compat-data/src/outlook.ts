export default {
  clients: [
    {
      name: 'Outlook 2000',
      aliases: ['outlook', 'outlook.legacy', 'outlook.win'],
      kind: 'application',
      devices: ['desktop'],
      os: 'windows',
      prefix: 'ms',
      versions: [
        {
          version: 9,
          released: '1999-06-27',
        },
      ],
    },
    {
      name: 'Outlook 2002',
      aliases: ['outlook', 'outlook.legacy', 'outlook.win'],
      kind: 'application',
      devices: ['desktop'],
      os: 'windows',
      prefix: 'ms',
      versions: [
        {
          version: 10,
          released: '2001-05-31',
          notes: 'Included in Office XP',
        },
      ],
    },
    {
      name: 'Outlook 2003',
      aliases: ['outlook', 'outlook.legacy', 'outlook.win'],
      kind: 'application',
      devices: ['desktop'],
      os: 'windows',
      prefix: 'ms',
      versions: [
        {
          version: 11,
          release: '2003-11-20',
          notes:
            'Included in Office 2003 and bundled with Exchange Server 2003',
        },
      ],
    },
    {
      name: 'Outlook 2007',
      aliases: ['outlook', 'outlook.curr'],
      kind: 'application',
      devices: ['desktop'],
      os: 'windows',
      prefix: 'mso',
      versions: [
        {
          version: 12,
          release: '2007-01-27',
          notes:
            'Included in Office 2007, except Office Home and Student edition',
        },
      ],
    },
    {
      name: 'Outlook 2010',
      aliases: ['outlook', 'outlook.curr'],
      kind: 'application',
      devices: ['desktop'],
      os: 'windows',
      prefix: 'mso',
      versions: [
        {
          version: 14,
          release: '2010-07-15',
          notes:
            'Included in Office 2010 Home and Business, Standard, Professional and Professional Plus',
        },
      ],
    },
    {
      name: 'Outlook 2013',
      aliases: ['outlook', 'outlook.curr'],
      kind: 'application',
      devices: ['desktop'],
      os: 'windows',
      prefix: 'mso',
      versions: [
        {
          version: 15,
          release: '2013-01-29',
          notes: 'Included in Office 2013, except Home & Student edition',
        },
      ],
    },
    {
      name: 'Outlook 2016',
      aliases: ['outlook', 'outlook.curr'],
      kind: 'application',
      devices: ['desktop'],
      os: 'windows',
      prefix: 'mso',
      versions: [
        {
          version: 16,
          release: '2015-09-22',
          notes: 'Included in Office 2016 and Office 365',
        },
      ],
    },
    {
      name: 'Outlook 2019',
      aliases: ['outlook', 'outlook.curr'],
      kind: 'application',
      devices: ['desktop'],
      os: 'windows',
      prefix: 'mso',
      versions: [
        {
          version: 16,
          release: '2018-09-24',
          notes: 'Included in Office 2019 and Office 365',
        },
      ],
    },
  ],
};
