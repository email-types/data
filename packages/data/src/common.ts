export type Link = {
  title: 'Microsoft' | 'MDN' | 'W3C' | 'Wikipedia' | (string & {});
  url: string;
};

/* Zero idea. Likely because it has not been tested */
export type Unknown = 'Unknown';
