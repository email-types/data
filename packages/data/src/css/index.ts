import dataTypes, { DataType, DataTypes } from './data-types';

export { dataTypes, DataType, DataTypes };
