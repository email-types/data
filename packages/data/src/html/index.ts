export * from './doctypes';
export * from './doctypes.types';
