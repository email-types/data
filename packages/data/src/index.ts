import * as css from './css';
import * as html from './html';
import * as mso from './mso';

export { css, html, mso };
