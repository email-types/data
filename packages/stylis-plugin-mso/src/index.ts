import { alternatives } from '@email-types/data/mso';

export type Plugin = (context: number, content: string) => string | void;

const properties = Object.keys(alternatives).reduce<Record<string, string>>(
  (result, property) => {
    // split the mso alt property to get the matched css style
    // eg. `mso-line-height-alt` becomes `line-height`.
    const [, name] = property.split(/mso-|-alt/);
    result[name] = property;
    return result;
  },
  {},
);

export const createPlugin = (
  options: { prefix: boolean } = { prefix: true },
): Plugin => {
  const plugin: Plugin = (context, content) => {
    // only run on a property declaration
    if (context === 1) {
      if (typeof options.prefix === 'boolean' && options.prefix !== false) {
        const [key, value] = content.trim().split(':');
        const property = properties[key];
        if (property) {
          return `${property}:${value};${content}`;
        }
      }

      // always fix the prefix
      return content.replace(/^-mso-/, 'mso-');
    }
  };

  return plugin;
};

export default createPlugin({ prefix: true });
