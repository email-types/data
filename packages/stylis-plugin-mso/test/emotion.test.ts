import renderer from 'react-test-renderer';
import createCache from '@emotion/cache';
import { CacheProvider, jsx, css, SerializedStyles } from '@emotion/core';
import { matchers } from 'jest-emotion';
import { createPlugin } from '../src';

expect.extend(matchers);

const render = (
  styles: SerializedStyles,
  prefix = true,
): renderer.ReactTestRendererJSON | null => {
  const cache = createCache({
    stylisPlugins: [createPlugin({ prefix })],
    prefix,
  });

  const child = jsx('div', { css: styles });
  const comp = jsx(CacheProvider, { value: cache }, child);

  return renderer.create(comp).toJSON();
};

describe('stylis-plugin-mso', () => {
  it('returns correct styles with mso prefix', () => {
    const tree = render(css({ paddingRight: 10 }));

    expect(tree).toHaveStyleRule('padding-right', '10px');
    expect(tree).toHaveStyleRule('mso-padding-right-alt', '10px');
    expect(tree).toMatchSnapshot();
  });

  it('returns correct styles without mso prefix', () => {
    const tree = render(css({ paddingLeft: 10 }), false);

    expect(tree).toHaveStyleRule('padding-left', '10px');
    expect(tree).not.toHaveStyleRule('mso-padding-left-alt', '10px');
    expect(tree).toMatchSnapshot();
  });

  it('returns correct styles with mso prefix when using string literal', () => {
    const tree = render(
      css`
        margin-left: 10px;
      `,
    );

    expect(tree).toHaveStyleRule('margin-left', '10px');
    expect(tree).toHaveStyleRule('mso-margin-left-alt', '10px');
    expect(tree).toMatchSnapshot();
  });
});
