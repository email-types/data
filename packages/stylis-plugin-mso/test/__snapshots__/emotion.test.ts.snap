// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stylis-plugin-mso returns correct styles with mso prefix 1`] = `
.emotion-0 {
  mso-padding-right-alt: 10px;
  padding-right: 10px;
}

<div
  className="emotion-0"
/>
`;

exports[`stylis-plugin-mso returns correct styles with mso prefix when using string literal 1`] = `
.emotion-0 {
  mso-margin-left-alt: 10px;
  margin-left: 10px;
}

<div
  className="emotion-0"
/>
`;

exports[`stylis-plugin-mso returns correct styles without mso prefix 1`] = `
.emotion-0 {
  padding-left: 10px;
}

<div
  className="emotion-0"
/>
`;
